Dev 1.0.3 Target as Next Stable Release

ExtremeFiretop opened the pull request:
@Martinski4GitHub 

Dev 1.0.3 Target as Next Stable Release

1. Moved a function for cleanliness.
2. Added a new email notification for failed checksum
3. Decided to stop toggling LEDs while a Backupmon update is in progress

Martinski4GitHub commented on MerlinAU.sh:
```diff
@@ -2721,13 +2741,17 @@ Please manually update to version $minimum_supported_version or higher to use th
         # Check if BACKUPMON version is greater than or equal to 1.44
         if [ "$current_version" -ge "$required_version" ]; then
+            #Temporarily Reset LEDs for NVRAM Backup
+            _Reset_LEDs_
```
Good idea. We don't need blinking LEDs while doing the backup.

Martinski4GitHub commented on MerlinAU.sh:
```diff
@@ -2692,6 +2701,7 @@ Please manually update to version $minimum_supported_version or higher to use th
         if [ "$fw_sig" != "$dl_sig" ]; then
             Say "${REDct}**ERROR**${NOct}: Extracted firmware does not match the SHA256 signature!"
             _DoCleanUp_ 1
+            _SendEMailNotification_ FAILED_FW_CHECKSUM_STATUS
```
Good idea. If the checksum fails, notify the user via email.

Martinski4GitHub reviewed (dismissed):
Looks good to go.

ExtremeFiretop commented:
@Martinski4GitHub 

Okay I fixed/addressed the issue brought up here: https://www.snbforums.com/threads/introducing-merlinau-the-ultimate-firmware-auto-updater-addon.88577/post-891658

I am going to wait for your review because I did have a drink or two tonight and you may have a better way to match the files.
I tested this method though and it works! :)

Also addressed BACKUPMON nvram differently, I moved BACKUPMON before the blink function starts, so it doesn't pause midway through the blink cycle.

Have a goodnight buddy. Much love!

Martinski4GitHub commented on MerlinAU.sh:
```diff
@@ -2574,8 +2637,8 @@ Please manually update to version $minimum_supported_version or higher to use th
-        # Define the path to the log file
-        changelog_file="${FW_BIN_DIR}/Changelog-NG.txt"
+        #Files matching the patterns 'Changelog-NG.txt' or 'Changelog-386.txt'
+        changelog_file=$(find ${FW_BIN_DIR} -type f \( -name "Changelog-386.txt" -o -name "Changelog-NG.txt" \) -print | head -n 1)
```
The "**find**" cmd statement uses the "**-type**" argument which is not valid with the built-in "find" command. It's valid with the Entware "find" utility which not all users may have installed. Also, I prefer to search for the pattern "**Changelog-*.txt**" which will find possible future changes in the name such as "Changelog-388.txt" in addition to current file names.

Martinski4GitHub commented:
> @Martinski4GitHub
> 
> Okay I fixed/addressed the issue brought up here: https://www.snbforums.com/threads/introducing-merlinau-the-ultimate-firmware-auto-updater-addon.88577/post-891658
> 
I fixed the "find" cmd statement because it was using an argument that is not valid with the built-in "find" command. See my comment for more details.

>
> Also addressed BACKUPMON nvram differently, I moved BACKUPMON before the blink function starts, so it doesn't pause midway through the blink cycle. Tested and it works as well!
> 
Sounds good.

ExtremeFiretop commented:
> > @Martinski4GitHub
> > Okay I fixed/addressed the issue brought up here: https://www.snbforums.com/threads/introducing-merlinau-the-ultimate-firmware-auto-updater-addon.88577/post-891658
> 
> I fixed the "find" cmd statement in my PR because it was using an argument that is not valid with the built-in "find" command. See my comment for more details.
> 
> > Also addressed BACKUPMON nvram differently, I moved BACKUPMON before the blink function starts, so it doesn't pause midway through the blink cycle. Tested and it works as well!
> 
> Sounds good.

Oh buddy this is why I wait for your!

Let me test this and review, on the surface not only does it look better but it just makes more freaking sense your way! 🤣

ExtremeFiretop commented:
@Martinski4GitHub  

That was super slick, because your PR was merged before mine into dev , my PR just picked up yours and updates/syncs the changes.

I was thinking about conflicts but I'm syncing into main instead of dev so, zero conflicts!

ExtremeFiretop commented:
Yep, tested fully now, your changes make me much more comfortable merging this in.

@Martinski4GitHub moving forwards it's back to dev to dev changes first, this one was just a bit of a last moment squeeze in!
